components/movementsensor/wheeledodometry/wheeled_odometry_test.go

```diff
@@ -0,0 +1,73 @@
+// Package wheeledodometry implements an odometery estimate from an encoder wheeled base
+package wheeledodometry
+
+import (
+	"context"
+	"testing"
+
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+// var count = 0
+
+func createFakeLeftMotor() motor.Motor {
+	var count = 0
+	return &inject.Motor{
+		PropertiesFunc: func(ctx context.Context, extra map[string]interface{}) (motor.Properties, error) {
+			return motor.Properties{PositionReporting: true}, nil
+		},
+		PositionFunc: func(ctx context.Context, extra map[string]interface{}) (float64, error) {
+			count++
+			return float64(1), nil
+		},
+		ResetZeroPositionFunc: func(ctx context.Context, offset float64, extra map[string]interface{}) error { return nil },
+		GoToFunc:              func(ctx context.Context, rpm, position float64, extra map[string]interface{}) error { return nil },
+		GoForFunc:             func(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error { return nil },
+		StopFunc:              func(ctx context.Context, extra map[string]interface{}) error { return nil },
+		SetPowerFunc:          func(ctx context.Context, powerPct float64, extra map[string]interface{}) error { return nil },
+	}
+}
+
+func createFakeRightMotor() motor.Motor {
+	var count = 0
+	return &inject.Motor{
+		PropertiesFunc: func(ctx context.Context, extra map[string]interface{}) (motor.Properties, error) {
+			return motor.Properties{PositionReporting: true}, nil
+		},
+		PositionFunc: func(ctx context.Context, extra map[string]interface{}) (float64, error) {
+			count++
+			return float64(-1), nil
+		},
+		ResetZeroPositionFunc: func(ctx context.Context, offset float64, extra map[string]interface{}) error { return nil },
+		GoToFunc:              func(ctx context.Context, rpm, position float64, extra map[string]interface{}) error { return nil },
+		GoForFunc:             func(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error { return nil },
+		StopFunc:              func(ctx context.Context, extra map[string]interface{}) error { return nil },
+		SetPowerFunc:          func(ctx context.Context, powerPct float64, extra map[string]interface{}) error { return nil },
+	}
+}
+
+func MakeConfig(t *testing.T) *odometry {
+	o := &odometry{
+		lastLeftPos:        0,
+		lastRightPos:       0,
+		baseWidth:          300,
+		wheelCircumference: 200,
+		timeInterval:       0.5,
+	}
+
+	mLeft := createFakeLeftMotor()
+	mRight := createFakeRightMotor()
+
+	o.leftMotors = append(o.leftMotors, mLeft)
+	o.rightMotors = append(o.rightMotors, mRight)
+
+	return o
+}
+
+func TestMath(t *testing.T) {
```

Comment:
Can you add a test with multiple motors in the config?

Reply:
yes will do. I need to add more tests in general